eth/db/witness.py

```diff
@@ -0,0 +1,67 @@
+from typing import (
+    Collection,
+    Dict,
+    NamedTuple,
+    Set,
+    Tuple,
+)
+
+from eth_typing import (
+    Address,
+    Hash32,
+)
+
+from eth.abc import (
+    WitnessIndexAPI,
+)
+
+
+class AccountQueryTracker(NamedTuple):
+    did_query_bytecode: bool
```

Comment:
I don't know how this is going to be used, but a more accurate name for this might be `did_query_nonempty_bytecode`, this is only marked true if we tried to read the bytecode and also got an answer out.